app/questionnaire/questionnaire_schema.py

```diff
@@ -166,6 +188,27 @@ def is_view_submitted_response_enabled(self) -> bool:
         is_enabled: bool = schema.get("view_response", False)
         return is_enabled
 
+    def get_all_when_rules_section_dependencies_for_section(
+        self, section_id: str
+    ) -> set[str]:
+        all_section_dependencies = self.when_rules_section_dependencies_by_section.get(
+            section_id, set()
+        ).union(
+            self.when_rules_section_dependencies_by_section_for_progress_value_source.get(
```

Comment:
Since you call it twice in this method and cast a set on it on both maybe just assign it a var and use that twice?

Reply:
Are you referring to `self.when_rules_section_dependencies_by_section_for_progress_value_source`? I think the other reference is slightly different: `self.when_rules_block_dependencies_by_section_for_progress_value_source`

Reply:
That's probably what I meant originally.